### PR TITLE
Implement GetZoneByProviderID and GetZoneByNodeName for Rackspace

### PR DESCRIPTION
**Pull request by FengyunPan**
This is part of #50926
cc @wlan0

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/rackspace/rackspace.go [hunk 1]
```diff
@@ -558,14 +558,18 @@ func (os *Rackspace) GetZone() (cloudprovider.Zone, error) {
 // This is particularly useful in external cloud providers where the kubelet
 // does not initialize node data.
 func (os *Rackspace) GetZoneByProviderID(providerID string) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByProviderID not implemented")
+	glog.V(1).Infof("Current zone is %v", os.region)
```
Comment (roberthbailey) [inline pkg/cloudprovider/providers/rackspace/rackspace.go]: Are these log statements necessary? I can see how they are useful for initial debugging, but this code is pretty straightforward so I don't see why they should be checked in.

Comment (FengyunPan) [reply pkg/cloudprovider/providers/rackspace/rackspace.go]: Right, they seems unnecessary, remove them like gce and aws.

File: pkg/cloudprovider/providers/rackspace/rackspace.go [hunk 2]
```diff
@@ -549,23 +549,21 @@ func (os *Rackspace) Routes() (cloudprovider.Routes, bool) {
...
-
 	return cloudprovider.Zone{Region: os.region}, nil
 }
 
 // GetZoneByProviderID implements Zones.GetZoneByProviderID
 // This is particularly useful in external cloud providers where the kubelet
 // does not initialize node data.
 func (os *Rackspace) GetZoneByProviderID(providerID string) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByProviderID not implemented")
+	return cloudprovider.Zone{Region: os.region}, nil
```
Comment (wlan0) [inline pkg/cloudprovider/providers/rackspace/rackspace.go]: Please utilize providerID in these methods.

If providerID does not affect the region variable, then please add a comment detailing why

File: pkg/cloudprovider/providers/rackspace/rackspace.go [hunk 3]
```diff
@@ -549,23 +549,21 @@ func (os *Rackspace) Routes() (cloudprovider.Routes, bool) {
...
 }
 
 // GetZoneByNodeName implements Zones.GetZoneByNodeName
 // This is particularly useful in external cloud providers where the kubelet
 // does not initialize node data.
 func (os *Rackspace) GetZoneByNodeName(nodeName types.NodeName) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByNodeName not imeplemented")
+	return cloudprovider.Zone{Region: os.region}, nil
```
Comment (wlan0) [inline pkg/cloudprovider/providers/rackspace/rackspace.go]: Same as above. Use nodeName

File: pkg/cloudprovider/providers/rackspace/rackspace.go (hunk 2 above)
Comment (FengyunPan) [reply pkg/cloudprovider/providers/rackspace/rackspace.go]: Done.

File: pkg/cloudprovider/providers/rackspace/rackspace.go (hunk 3 above)
Comment (FengyunPan) [reply pkg/cloudprovider/providers/rackspace/rackspace.go]: Done.

File: pkg/cloudprovider/providers/rackspace/rackspace.go (hunk 2 above)
Comment (FengyunPan) [reply pkg/cloudprovider/providers/rackspace/rackspace.go]: I am sorry for my mistake, we should get zone from server by providerID or NodeName.

**Comment by FengyunPan** [comment]
@dims Since PR #51317 is merged, and this PR is similar to PR #51317, so can you approve this PR? 
PTAL, thank you.

**Comment by FengyunPan** [comment]
ping @mikedanese
ping @thockin
Can you approve this PR?

**Pull request closed without merging**